### PR TITLE
add flag for protocol sniffing for virtual inbound

### PR DESCRIPTION
**Pull request by yxue**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/util/util.go [hunk 1]
```diff
@@ -282,6 +282,11 @@ func IsProtocolSniffingEnabledForNode(node *model.Proxy) bool {
 	return features.EnableProtocolSniffing.Get() && IsIstioVersionGE13(node)
 }
 
+func IsProtocolSniffingEnabledForInbound(node *model.Proxy) bool {
+	return IsProtocolSniffingEnabledForNode(node) ||
```
Comment (howardjohn) [inline pilot/pkg/networking/util/util.go]: I don't think we want this.

We want to be able to set protocol sniffing for inbound only, or protocol sniffing for everything.

The way this is set up you can do protocol sniffing for outbound only, or everything

Comment (howardjohn) [reply pilot/pkg/networking/util/util.go]: disregard this.. I was thinking and not or

**Comment by howardjohn** [comment]
see https://github.com/istio/istio/issues/16869

**Comment by rlenglet** [comment]
`istio-unit-tests-release-1.3` failed with
```
=== RUN   TestVirtualInboundListenerBuilder
2019-09-06T20:28:03.020955Z	warn	Failed to parse IP Address from plugin listener
--- FAIL: TestVirtualInboundListenerBuilder (0.00s)
    listener_builder_test.go:201: found virtual listener: virtualOutbound
    listener_builder_test.go:207: found virtual inbound listener: virtualInbound
    listener_builder_test.go:224: expect virtual listener has 1 passthrough listeners, found 2
=== RUN   TestVirtualInboundHasPassthroughClusters
2019-09-06T20:28:03.022932Z	warn	Failed to parse IP Address from plugin listener
--- FAIL: TestVirtualInboundHasPassthroughClusters (0.00s)
    listener_builder_test.go:248: expect virtual listener has 4 filter chains as the sum of 2nd level listeners plus the 1 fallthrough filter chains, found 4
```

File: pilot/pkg/features/pilot.go [hunk 2]
```diff
@@ -183,6 +183,12 @@ var (
 		"If enabled, protocol sniffing will be used on ports whose port protocol is not specified or unsupported",
 	)
 
+	EnableProtocolSniffingForInbound = env.RegisterBoolVar(
+		"PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_INBOUND",
+		false,
```
Comment (howardjohn) [inline pilot/pkg/features/pilot.go]:
```suggestion
true
```

**Comment by howardjohn** [comment]
@rshriram please take a look. This has/will been updated to just be a different flag for inbound and outbound but both enabled by default

**Pull request merged**